### ISSUE TITLE
large_tuple violation false positive when async function is a method parameter

### ISSUE THREAD

**Issue by kmahar** [realm/SwiftLint#3753]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
When a function accepts an `async` function of 3+ parameters as a parameter, it is erroneously considered a violation of the `large_tuple` rule.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ echo "private func foo(bar: (Int, String, Float) async throws -> Void) {}" | swiftlint lint --no-cache --use-stdin --enable-all-rules
```

Output:
```
<nopath>:1:18: warning: Large Tuple Violation: Tuples should have at most 2 members. (large_tuple)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.45.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Built from source.
* Paste your configuration file: No config file, this reproduces with the default configuration options.

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No.
* Which Xcode version are you using (check `xcodebuild -version`)? I'm on Linux, so no Xcode version.
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
private func foo(bar: (Int, String, Float) async throws -> Void) {}
```
